chore: README updates (#2335)
hashicorp/terraform-website: README.md

## Where the Docs Live

This repository contains docs for Terraform Enterprise, located in [`/content/enterprise`](https://github.com/hashicorp/terraform-website/tree/master/content/enterprise).

> The rest of the docs are located in various external repositories:

Reviewer: ```suggestion
The rest of the content is located in the following repositories:
```

[plugin/log]: https://www.terraform.io/plugin/log
[plugin/mux]: https://www.terraform.io/plugin/mux
[plugin/sdkv2]: https://www.terraform.io/plugin/sdkv2
[registry]: https://www.terraform.io/registry

[terraform-cdk]: https://github.com/hashicorp/terraform-cdk
[terraform]: https://github.com/hashicorp/terraform
[terraform-website]: https://github.com/hashicorp/terraform-cdk
[terraform-docs-common]: https://github.com/hashicorp/terraform-docs-common
[terraform-docs-agents]: https://github.com/hashicorp/terraform-docs-agents
[terraform-plugin-sdk]: https://github.com/hashicorp/terraform-plugin-sdk
[terraform-plugin-log]: https://github.com/hashicorp/terraform-plugin-log
[terraform-plugin-mux]: https://github.com/hashicorp/terraform-plugin-mux
[terraform-plugin-framework]: https://github.com/hashicorp/terraform-plugin-framework

  **Notable branches:** `master` is the "live" content that gets deployed to [terraform.io](https://terraform.io). The site gets redeployed for new commits to master.

## Deploying Changes to [terraform.io](https://terraform.io)

### For changes in this repo

Merge the PR to master, and the site will automatically deploy in about 5m. 🙌

> ### For changes in any other listed repo

Reviewer: ```suggestion
### For changes in any other listed repositories
```

There are various GitHub refs that will cause docs content to be deployed to terraform.io,

> and these may vary on a repo-to-repo basis but generally speaking, these are limited to `stable-website`, `main`, and a few others.

Reviewer: If we're not going to mention the others, maybe just stick with how the latest docs are updated? `and a few others` isn't too helpful 😄

> and these may vary on a repo-to-repo basis but generally speaking, these are limited to `stable-website`, `main`, and a few others.

Author: @BRKalow ☝️ ?

> and these may vary on a repo-to-repo basis but generally speaking, these are limited to `stable-website`, `main`, and a few others.

Reviewer: LGTM!

> Changes will be deployed to terraform.io roughly every hour

Reviewer: ```suggestion
Changes will be deployed to terraform.io roughly every hour.
```

## Where the Docs Live

> As of **May 25, 2022**, this repository only houses docs for Terraform Enterprise.
> This will also change soon.

Reviewer: I don't think we need the extra context here:
```suggestion
This repository contains docs for Terraform Enterprise, located in [`/content/enterprise`](https://github.com/hashicorp/terraform-website/tree/master/content/enterprise).
```
